src/AddIns/Uno.WinUI.Graphics3DGL/Uno.WinUI.Graphics3DGL.csproj

```diff
@@ -66,17 +66,23 @@
 		<DefaultItemExcludes>$(DefaultItemExcludes);angle/**</DefaultItemExcludes>
 	</PropertyGroup>
 
-	<Target Name="GetAngle" BeforeTargets="CoreCompile">
+	<Target Name="GetAngle" BeforeTargets="Build">
 		<DownloadFile
+			Condition="!exists('$(MSBuildThisFileDirectory)angle_binaries/osx/libEGL.dylib')"
 			SourceUrl="https://github.com/dotnet/Silk.NET/raw/a9c75a61528ee7b4e55442f5141cb132f10d72ff/src/Native/Silk.NET.OpenGLES.ANGLE.Native/runtimes/osx/native/libEGL.dylib"
 			DestinationFolder="$(MSBuildThisFileDirectory)angle_binaries/osx">
 		</DownloadFile>
 		<DownloadFile
+			Condition="!exists('$(MSBuildThisFileDirectory)angle_binaries/osx/libGLESv2.dylib')"
 			SourceUrl="https://github.com/dotnet/Silk.NET/raw/a9c75a61528ee7b4e55442f5141cb132f10d72ff/src/Native/Silk.NET.OpenGLES.ANGLE.Native/runtimes/osx/native/libGLESv2.dylib"
 			DestinationFolder="$(MSBuildThisFileDirectory)angle_binaries/osx">
 		</DownloadFile>
 	</Target>
 
+	<Target Name="CleanAngle" AfterTargets="Clean">
+		<RemoveDir Directories="$(MSBuildThisFileDirectory)angle_binaries" />
+	</Target>
```

Comment:
That's likely not the right way. The right way is to do `<FileWrites Include="angle_binaries/**" />` in an `ItemGroup` just after the download task.

Reply:
Let's fix that in a separate PR.